src/transformers/generation/utils.py

```diff
@@ -1594,6 +1592,20 @@ def generate(
                 )
             if generation_config.cache_implementation == "static":
                 model_kwargs["past_key_values"] = self._get_static_cache(batch_size, generation_config.max_length)
+            elif generation_config.cache_implementation == "quantized":
+                if not is_quanto_available():
+                    raise ImportError(
+                        "You need to install `quanto` in order to use KV cache quantization. "
+                        "Please install it via  with `pip install git+https://github.com/huggingface/quanto`"
+                    )
+
+                cache_config = (
+                    generation_config.cache_config
+                    if generation_config.cache_config is not None
+                    else QuantizedCacheConfig()
+                )
+
+                model_kwargs["past_key_values"] = QuantoQuantizedCache(**cache_config.to_dict())
```

Comment:
```suggestion
                model_kwargs["past_key_values"] = QuantoQuantizedCache(cache_config)
```

Reply:
Not really, the class does not accept as input a `dataclass` object so we pass it as **mapping

Reply:
ah yeah, it's annoying